src/Stratis.Bitcoin/Configuration/NodeSettings.cs

```diff
@@ -116,86 +125,110 @@ public NodeSettings(Network innerNetwork = null, ProtocolVersion protocolVersion
+                return this;
+
+            var args = this.LoadArgs;
+
+            // Ensure that we have a configuration file
+            var fileArgs = (this.ConfigurationFile != null && File.Exists(this.ConfigurationFile)) ? 
+                File.ReadAllText(this.ConfigurationFile) : 
+                this.DefaultConfigurationFile();
+
             // Find out if we need to run on testnet or regtest from the config file.
-            if (this.ConfigurationFile != null)
-            {
-                AssertConfigFileExists(this.ConfigurationFile);
-                var configTemp = new TextFileConfiguration(File.ReadAllText(this.ConfigurationFile));
-                this.Testnet = configTemp.GetOrDefault<bool>("testnet", false);
-                this.RegTest = configTemp.GetOrDefault<bool>("regtest", false);
-            }
+            var config = new TextFileConfiguration(fileArgs);
+            this.Testnet = config.GetOrDefault<bool>("testnet", false);
+            this.RegTest = config.GetOrDefault<bool>("regtest", false);
 
-            //Only if args contains -testnet, do we set it to true, otherwise it overwrites file configuration
+            // Only if args contains -testnet, do we set it to true, otherwise it overwrites file configuration
             if (args.Contains("-testnet", StringComparer.CurrentCultureIgnoreCase))
                 this.Testnet = true;
 
-            //Only if args contains -regtest, do we set it to true, otherwise it overwrites file configuration
+            // Only if args contains -regtest, do we set it to true, otherwise it overwrites file configuration
             if (args.Contains("-regtest", StringComparer.CurrentCultureIgnoreCase))
                 this.RegTest = true;
 
             if (this.Testnet && this.RegTest)
                 throw new ConfigurationException("Invalid combination of -regtest and -testnet.");
 
+            var consoleConfig = new TextFileConfiguration(args);
+            consoleConfig.MergeInto(config);
+            this.ConfigReader = config;
+
             this.Network = this.GetNetwork();
-            
+
             // Setting the data directory.
-            if (dataDir == null)
+            if (this.DataDir == null)
             {
                 this.DataDir = this.CreateDefaultDataDirectories(Path.Combine("StratisNode", this.Network.RootFolderName), this.Network);
-            }
-            else
-            {
-                // Create the data directories if they don't exist.
-                string directoryPath = Path.Combine(dataDir, this.Network.RootFolderName, this.Network.Name);
-                Directory.CreateDirectory(directoryPath);
-                this.DataDir = directoryPath;
-                this.Logger.LogDebug("Data directory initialized with path {0}.", directoryPath);
-            }
-            
-            // If no configuration file path is passed in the args, load the default file.
-            if (this.ConfigurationFile == null)
```

Comment:
PS why did this logic change here? I see that we did check if the ConfigFile is null and if so, created a default one?

Reply:
The new logic is still doing that. The old logic was not very consistent between the creation of the configuration file and actually using the settings within it.